packages/slice-machine/src/domain/fields.ts

```diff
@@ -196,6 +197,18 @@ export const UIDField: UIDField = {
   type: "UID",
 };
 
+export const UIDFieldLabelSchema = z.string().max(35).min(1);
+
+export const UIDFieldCustomErrorMap: z.ZodErrorMap = (issue, ctx) => {
+  if (issue.code === z.ZodIssueCode.too_big) {
+    return { message: `String is too long. Max: ${issue.maximum}` };
+  }
+  if (issue.code === z.ZodIssueCode.too_small) {
+    return { message: "This field is required" };
+  }
+  return { message: ctx.defaultError };
+};
+
```

Comment:
I'd move this to the form module, as a private function - i would also:
- encapsulate the schema parsing in a function so you don't have to call parse and pass the map
- use switch for the error map implementation
- provide a default error message that is controlled by us, because we don't know what zod spits out in the default case
- use more consistent punctuation in the sentences, for example, the first can be "The value can't be longer than x characters`